src/appleseed-max-impl/utilities.cpp

```diff
@@ -31,6 +31,7 @@
 
 // appleseed-max headers.
 #include "appleseedoslplugin/osltexture.h"
+#include "appleseedrenderer/appleseedrenderer.h"
```

Comment:
I find it a bit concerning that `utilities` depends on `appleseedrenderer`: lower level components should never depend on higher level ones. Maybe `get_current_time()` should be declared in `appleseedrenderer.{h,cpp}`.

For the same reason I'm slightly concerned about the dependencies toward `appleseedoslplugin/osltexture.h` and `osloutputselectormap/osloutputselector.h`.

Reply:
Can I make `get_current_time` in appleseedrenderer a global function, not the class member? So that I don't have to check and cast current renderer to get the time.

Reply:
Yes, that's what I meant. You could also make it a static method of `AppleseedRenderer` to highlight the dependency to that class.

Reply:
I'm trying to move it to the renderer class but in some utilities I still need to call that function so I still need to include renderer header.

Reply:
Other way is to pass `time` from the renderer to all that `create_material` and so on as an argument.